File: app/src/main/java/com/katkov/lolachievements/data/mappers/SummonerDTOApiToDomainMapper.java
package com.katkov.lolachievements.data.mappers;

import com.katkov.lolachievements.data.cloud.model.SummonerDTOApiModel;
import com.katkov.lolachievements.domain.model.SummonerDTO;

import javax.inject.Inject;

public class SummonerDTOApiToDomainMapper extends Mapper<SummonerDTOApiModel, SummonerDTO> {

    @Inject
    public SummonerDTOApiToDomainMapper() {
    }

    @Override
    public SummonerDTO map(SummonerDTOApiModel source) {
        return new SummonerDTO(
                source.getName(),
                source.getSummonerLevel(),
                source.getId());
    }
}

File: app/src/main/java/com/katkov/lolachievements/data/mappers/SummonerDTOApiToDomainMapper.kt
package com.katkov.lolachievements.data.mappers

import com.katkov.lolachievements.data.cloud.model.SummonerDTOApiModel
import com.katkov.lolachievements.domain.model.SummonerDTO

import javax.inject.Inject

class SummonerDTOApiToDomainMapper
@Inject
constructor() : Mapper<SummonerDTOApiModel, SummonerDTO>() {

    override fun map(source: SummonerDTOApiModel): SummonerDTO {
        return SummonerDTO(
                source.name,
                source.summonerLevel,
                source.id)
    }
}
